cloud/deployment/inspect/inspect.go

```diff
@@ -0,0 +1,157 @@
+package inspect
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/astronomer/astro-cli/astro-client"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/astronomer/astro-cli/config"
+)
+
+type deploymentInfo struct {
+	DeploymentID   string    `mapstructure:"deployment-id"`
+	WorkspaceID    string    `mapstructure:"workspace-id"`
+	ClusterID      string    `mapstructure:"cluster-id"`
+	AirflowVersion string    `mapstructure:"airflow-version"`
+	ReleaseName    string    `mapstructure:"release-name"`
+	DeploymentURL  string    `mapstructure:"deployment-url"`
+	WebserverURL   string    `mapstructure:"webserver-url"`
+	UpdatedAt      time.Time `mapstructure:"updated-at"`
+	Status         string    `mapstructure:"status"`
+}
+
+type deploymentConfig struct {
+	Name              string              `mapstructure:"name"`
+	Description       string              `mapstructure:"description"`
+	ClusterID         string              `mapstructure:"cluster-id"` // this is also in deploymentInfo
+	RunTimeVersion    string              `mapstructure:"runtime-version"`
+	SchedulerAU       int                 `mapstructure:"scheduler-au"`
+	SchedulerReplicas int                 `mapstructure:"scheduler-replicas"`
+	AlertEmails       []string            `mapstructure:"alert-emails"`
+	WorkerQueues      []astro.WorkerQueue `mapstructure:"worker-queues"`
+	AstroVariables    Variables           `mapstructure:"astronomer-variables"`
+}
+
+type PrintableDeployment struct {
+	DeploymentInfo   map[string]interface{} `mapstructure:"deployment-info"`
+	DeploymentConfig map[string]interface{} `mapstructure:"deployment-config"`
+}
+
+type Variables struct {
+	EnvFromFile  string
+	VariableList []astro.EnvironmentVariablesObject `mapstructure:"astronomer-variables"`
+}
+
+var (
+	jsonMarshal = json.MarshalIndent
+	yamlMarshal = yaml.Marshal
+)
+
+const (
+	jsonFormat = "json"
+)
+
+func Inspect(wsID, deploymentName, deploymentID, outputFormat string, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment                    astro.Deployment
+		err                                    error
+		infoToPrint                            []byte
+		deploymentInfoMap, deploymentConfigMap map[string]interface{}
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(wsID, deploymentID, deploymentName, client)
```

Comment:
why are we still using this method over `client.GetDeployment`? I mean someday this method is not going to scale and we would need to get rid of it or end up in a bad tech debt. Also, this would mean we should only offer users to pass `deploymentID` and not `deploymentName`.

Reply:
> but we could get similar feedback from users that try this out as well.

IMO deprecating a flag once it is released will be much harder compared to deciding it upfront

Reply:
I will talk through it with @sunkickr one more time.

Reply:
@neel-astro in terms of `deploymentName` vs. `deploymentID`, we are eventually going to move everything to use `deploymentName` as that is what users are requesting. 

As far as the `deployment.GetDeployment()` goes that is an attempt to get a consistent UX for all deployment commands so that refactoring can be localized to one method and it will change the way deployment commands work in wholesale. For now we are going to keep the use of it. 

> this method is not going to scale and we would need to get rid of it or end up in a bad tech debt

However, I am curious to understand your comment above. The only request is let's make a new issue for it and discuss it there.